<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from dual_ur_robotiq_rs.srdf.xacro  | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="dual_ur_robotiq_rs">
  <!--GROUPS - Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS - When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS - When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS - When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS - Groups can also be formed by referencing to already defined group names-->
  <group name="left_ur_manipulator">
    <chain base_link="left_base_link" tip_link="left_tool0"/>
  </group>
  <!--GROUP STATES - Purpose - Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state group="left_ur_manipulator" name="left_home">
    <joint name="left_elbow_joint" value="0"/>
    <joint name="left_shoulder_lift_joint" value="-1.5707"/>
    <joint name="left_shoulder_pan_joint" value="0"/>
    <joint name="left_wrist_1_joint" value="0"/>
    <joint name="left_wrist_2_joint" value="0"/>
    <joint name="left_wrist_3_joint" value="0"/>
  </group_state>
  <group_state group="left_ur_manipulator" name="left_up">
    <joint name="left_elbow_joint" value="0"/>
    <joint name="left_shoulder_lift_joint" value="-1.5707"/>
    <joint name="left_shoulder_pan_joint" value="0"/>
    <joint name="left_wrist_1_joint" value="-1.5707"/>
    <joint name="left_wrist_2_joint" value="0"/>
    <joint name="left_wrist_3_joint" value="0"/>
  </group_state>
  <group_state group="left_ur_manipulator" name="left_test_configuration">
    <joint name="left_elbow_joint" value="1.4"/>
    <joint name="left_shoulder_lift_joint" value="-1.62"/>
    <joint name="left_shoulder_pan_joint" value="1.54"/>
    <joint name="left_wrist_1_joint" value="-1.2"/>
    <joint name="left_wrist_2_joint" value="-1.6"/>
    <joint name="left_wrist_3_joint" value="-0.11"/>
  </group_state>
  <!--END EFFECTOR - Purpose - Represent information about an end effector.-->
  <!--VIRTUAL JOINT - Purpose - this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
  <!--DISABLE COLLISIONS - By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <disable_collisions link1="left_base_link" link2="left_base_link_inertia" reason="Adjacent"/>
  <disable_collisions link1="left_base_link_inertia" link2="left_shoulder_link" reason="Adjacent"/>
  <disable_collisions link1="left_tool0" link2="left_wrist_1_link" reason="Never"/>
  <disable_collisions link1="left_tool0" link2="left_wrist_2_link" reason="Never"/>
  <disable_collisions link1="left_tool0" link2="left_wrist_3_link" reason="Adjacent"/>
  <disable_collisions link1="left_forearm_link" link2="left_upper_arm_link" reason="Adjacent"/>
  <disable_collisions link1="left_forearm_link" link2="left_wrist_1_link" reason="Adjacent"/>
  <disable_collisions link1="left_shoulder_link" link2="left_upper_arm_link" reason="Adjacent"/>
  <disable_collisions link1="left_wrist_1_link" link2="left_wrist_2_link" reason="Adjacent"/>
  <disable_collisions link1="left_wrist_1_link" link2="left_wrist_3_link" reason="Never"/>
  <disable_collisions link1="left_wrist_2_link" link2="left_wrist_3_link" reason="Adjacent"/>
  <group name="left_robotiq_2f_85_gripper">
    <link name="left_robotiq_85_base_link"/>
    <link name="left_robotiq_85_left_inner_knuckle_link"/>
    <link name="left_robotiq_85_left_knuckle_link"/>
    <link name="left_robotiq_85_left_finger_link"/>
    <link name="left_robotiq_85_left_finger_tip_link"/>
    <link name="left_robotiq_85_right_inner_knuckle_link"/>
    <link name="left_robotiq_85_right_knuckle_link"/>
    <link name="left_robotiq_85_right_finger_link"/>
    <link name="left_robotiq_85_right_finger_tip_link"/>
  </group>
  <!--GROUP STATES - Purpose - Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state group="left_robotiq_2f_85_gripper" name="left_open">
    <joint name="left_robotiq_85_left_knuckle_joint" value="0"/>
  </group_state>
  <group_state group="left_robotiq_2f_85_gripper" name="left_close">
    <joint name="left_robotiq_85_left_knuckle_joint" value="0.7"/>
  </group_state>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_left_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_left_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_left_inner_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_left_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_robotiq_85_right_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_left_finger_tip_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_left_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_left_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_left_inner_knuckle_link" reason="Default"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_left_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_robotiq_85_left_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_knuckle_link" link2="left_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_knuckle_link" link2="left_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_knuckle_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_knuckle_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_finger_link" link2="left_robotiq_85_right_finger_tip_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_right_finger_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_finger_link" link2="left_robotiq_85_right_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_right_finger_tip_link" link2="left_robotiq_85_right_inner_knuckle_link" reason="Default"/>
  <disable_collisions link1="left_robotiq_85_right_finger_tip_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_inner_knuckle_link" link2="left_robotiq_85_right_knuckle_link" reason="Never"/>
  <group name="ur_robotiq">
    <group name="ur_manipulator"/>
    <group name="robotiq_2f_85_gripper"/>
  </group>
  <!--DISABLE COLLISIONS By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <disable_collisions link1="left_robotiq_85_base_link" link2="left_ur_to_robotiq_link" reason="Adjacent"/>
  <disable_collisions link1="left_robotiq_85_left_finger_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_finger_tip_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_inner_knuckle_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_left_knuckle_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_finger_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_finger_tip_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_inner_knuckle_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_robotiq_85_right_knuckle_link" link2="left_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="left_ur_to_robotiq_link" link2="left_wrist_1_link" reason="Never"/>
  <disable_collisions link1="left_ur_to_robotiq_link" link2="left_wrist_2_link" reason="Never"/>
  <disable_collisions link1="left_ur_to_robotiq_link" link2="left_wrist_3_link" reason="Adjacent"/>
  <!--GROUPS - Representation of a set of joints and links. This can be useful for specifying DOF to plan for, defining arms, end effectors, etc-->
  <!--LINKS - When a link is specified, the parent joint of that link (if it exists) is automatically included-->
  <!--JOINTS - When a joint is specified, the child link of that joint (which will always exist) is automatically included-->
  <!--CHAINS - When a chain is specified, all the links along the chain (including endpoints) are included in the group. Additionally, all the joints that are parents to included links are also included. This means that joints along the chain and the parent joint of the base link are included in the group-->
  <!--SUBGROUPS - Groups can also be formed by referencing to already defined group names-->
  <group name="right_ur_manipulator">
    <chain base_link="right_base_link" tip_link="right_tool0"/>
  </group>
  <!--GROUP STATES - Purpose - Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state group="right_ur_manipulator" name="right_home">
    <joint name="right_elbow_joint" value="0"/>
    <joint name="right_shoulder_lift_joint" value="-1.5707"/>
    <joint name="right_shoulder_pan_joint" value="0"/>
    <joint name="right_wrist_1_joint" value="0"/>
    <joint name="right_wrist_2_joint" value="0"/>
    <joint name="right_wrist_3_joint" value="0"/>
  </group_state>
  <group_state group="right_ur_manipulator" name="right_up">
    <joint name="right_elbow_joint" value="0"/>
    <joint name="right_shoulder_lift_joint" value="-1.5707"/>
    <joint name="right_shoulder_pan_joint" value="0"/>
    <joint name="right_wrist_1_joint" value="-1.5707"/>
    <joint name="right_wrist_2_joint" value="0"/>
    <joint name="right_wrist_3_joint" value="0"/>
  </group_state>
  <group_state group="right_ur_manipulator" name="right_test_configuration">
    <joint name="right_elbow_joint" value="1.4"/>
    <joint name="right_shoulder_lift_joint" value="-1.62"/>
    <joint name="right_shoulder_pan_joint" value="1.54"/>
    <joint name="right_wrist_1_joint" value="-1.2"/>
    <joint name="right_wrist_2_joint" value="-1.6"/>
    <joint name="right_wrist_3_joint" value="-0.11"/>
  </group_state>
  <!--END EFFECTOR - Purpose - Represent information about an end effector.-->
  <!--VIRTUAL JOINT - Purpose - this element defines a virtual joint between a robot link and an external frame of reference (considered fixed with respect to the robot)-->
  <!--DISABLE COLLISIONS - By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <disable_collisions link1="right_base_link" link2="right_base_link_inertia" reason="Adjacent"/>
  <disable_collisions link1="right_base_link_inertia" link2="right_shoulder_link" reason="Adjacent"/>
  <disable_collisions link1="right_tool0" link2="right_wrist_1_link" reason="Never"/>
  <disable_collisions link1="right_tool0" link2="right_wrist_2_link" reason="Never"/>
  <disable_collisions link1="right_tool0" link2="right_wrist_3_link" reason="Adjacent"/>
  <disable_collisions link1="right_forearm_link" link2="right_upper_arm_link" reason="Adjacent"/>
  <disable_collisions link1="right_forearm_link" link2="right_wrist_1_link" reason="Adjacent"/>
  <disable_collisions link1="right_shoulder_link" link2="right_upper_arm_link" reason="Adjacent"/>
  <disable_collisions link1="right_wrist_1_link" link2="right_wrist_2_link" reason="Adjacent"/>
  <disable_collisions link1="right_wrist_1_link" link2="right_wrist_3_link" reason="Never"/>
  <disable_collisions link1="right_wrist_2_link" link2="right_wrist_3_link" reason="Adjacent"/>
  <group name="right_robotiq_2f_85_gripper">
    <link name="right_robotiq_85_base_link"/>
    <link name="right_robotiq_85_left_inner_knuckle_link"/>
    <link name="right_robotiq_85_left_knuckle_link"/>
    <link name="right_robotiq_85_left_finger_link"/>
    <link name="right_robotiq_85_left_finger_tip_link"/>
    <link name="right_robotiq_85_right_inner_knuckle_link"/>
    <link name="right_robotiq_85_right_knuckle_link"/>
    <link name="right_robotiq_85_right_finger_link"/>
    <link name="right_robotiq_85_right_finger_tip_link"/>
  </group>
  <!--GROUP STATES - Purpose - Define a named state for a particular group, in terms of joint values. This is useful to define states like 'folded arms'-->
  <group_state group="right_robotiq_2f_85_gripper" name="right_open">
    <joint name="right_robotiq_85_left_knuckle_joint" value="0"/>
  </group_state>
  <group_state group="right_robotiq_2f_85_gripper" name="right_close">
    <joint name="right_robotiq_85_left_knuckle_joint" value="0.7"/>
  </group_state>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_left_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_left_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_left_inner_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_left_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_robotiq_85_right_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_left_finger_tip_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_left_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_left_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_left_inner_knuckle_link" reason="Default"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_left_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_robotiq_85_left_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_knuckle_link" link2="right_robotiq_85_right_finger_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_knuckle_link" link2="right_robotiq_85_right_finger_tip_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_knuckle_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_knuckle_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_finger_link" link2="right_robotiq_85_right_finger_tip_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_right_finger_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_finger_link" link2="right_robotiq_85_right_knuckle_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_right_finger_tip_link" link2="right_robotiq_85_right_inner_knuckle_link" reason="Default"/>
  <disable_collisions link1="right_robotiq_85_right_finger_tip_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_inner_knuckle_link" link2="right_robotiq_85_right_knuckle_link" reason="Never"/>
  <group name="ur_robotiq">
    <group name="ur_manipulator"/>
    <group name="robotiq_2f_85_gripper"/>
  </group>
  <!--DISABLE COLLISIONS By default it is assumed that any link of the robot could potentially come into collision with any other link in the robot. This tag disables collision checking between a specified pair of links. -->
  <disable_collisions link1="right_robotiq_85_base_link" link2="right_ur_to_robotiq_link" reason="Adjacent"/>
  <disable_collisions link1="right_robotiq_85_left_finger_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_finger_tip_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_inner_knuckle_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_left_knuckle_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_finger_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_finger_tip_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_inner_knuckle_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_robotiq_85_right_knuckle_link" link2="right_ur_to_robotiq_link" reason="Never"/>
  <disable_collisions link1="right_ur_to_robotiq_link" link2="right_wrist_1_link" reason="Never"/>
  <disable_collisions link1="right_ur_to_robotiq_link" link2="right_wrist_2_link" reason="Never"/>
  <disable_collisions link1="right_ur_to_robotiq_link" link2="right_wrist_3_link" reason="Adjacent"/>
  <group name="both_manipulators">
    <group name="left_ur_manipulator"/>
    <group name="right_ur_manipulator"/>
  </group>
</robot>
